classes/galley/Galley.inc.php

```diff
@@ -0,0 +1,232 @@
+<?php
+
+/**
+ * @file classes/galley/Galley.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2003-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class galley
+ * @ingroup galley
+ *
+ * @brief A galley is a presentation version of a published submission file.
+ */
+
+namespace PKP\galley;
+
+use APP\core\Application;
+
+use APP\facades\Repo;
+use PKP\facades\Locale;
+use PKP\submission\Representation;
+use PKP\submissionFile\SubmissionFile;
+
+class Galley extends Representation
+{
+    public ?SubmissionFile $_submissionFile = null;
+
+    //
+    // Get/set methods
+    //
+    /**
+     * Get views count.
+     *
+     * @return int
+     */
+    public function getViews()
+    {
+        $application = Application::get();
+        $fileId = $this->getFileId();
+        if ($fileId) {
+            return $application->getPrimaryMetricByAssoc(ASSOC_TYPE_SUBMISSION_FILE, $fileId);
+        } else {
+            return 0;
+        }
+    }
+
+    /**
+     * Get label/title.
+     *
+     * @return string
+     */
+    public function getLabel()
+    {
+        return $this->getData('label');
+    }
+
+    /**
+     * Set label/title.
+     *
+     * @param $label string
+     */
+    public function setLabel($label)
+    {
+        $this->setData('label', $label);
+    }
+
+    /**
+     * Get locale.
+     *
+     * @return string
+     */
+    public function getLocale()
+    {
+        return $this->getData('locale');
+    }
+
+    /**
+     * Set locale.
+     *
+     * @param $locale string
+     */
+    public function setLocale($locale)
+    {
+        $this->setData('locale', $locale);
+    }
+
+    /**
+     * Return the "best" article ID -- If a public article ID is set,
+     * use it; otherwise use the internal article Id.
+     *
+     * @return string
+     */
+    public function getBestGalleyId()
+    {
+        return $this->getData('urlPath')
+            ? $this->getData('urlPath')
+            : $this->getId();
+    }
+
+    /**
+     * Set file ID.
+     *
+     * @deprecated 3.3
+     *
+     * @param $fileId int
+     */
+    public function setFileId($fileId)
```

Comment:
I don't mind if we remove these deprecated functions, since they were deprecated in 3.3 and this is now being rewritten.

Reply:
I've removed `getFileId()` and `setFileId()`. I started working on removing `getFile()`, but it turned into a bit of a struggle. It's used in some tricky places, like template code that does:

```
{foreach from="$article->getGalleys()" item=galley}
	{assign var="file" value=$galley->getFile()}
	...
{/foreach}
```

There are similar places where `getFileType()` are used, which relies on `getFile()`. We should refactor away from those, but it will take some doing to restructure how data is getting to templates (`article_summary.tpl` is a particularly tricky one). In the interests of time, I'd prefer to defer that work. Ok?

Reply:
Yes, that's OK with me; thanks for looking into it.